plugins/modules/foreman_hostgroup.py

```diff
@@ -408,12 +414,34 @@ def main():
 
     parameters = entity_dict.get('parameters')
 
+    puppetclasses = entity_dict.pop('puppetclasses', None)
+    current_puppetclasses = entity.pop('puppetclasses', None) if entity else []
+
     hostgroup = module.ensure_entity('hostgroups', entity_dict, entity)
 
     if hostgroup:
         scope = {'hostgroup_id': hostgroup['id']}
         module.ensure_scoped_parameters(scope, entity, parameters)
 
+    if not module.desired_absent and 'environment_id' in hostgroup:
+        if puppetclasses is not None:
+            current_puppetclasses = [p['id'] for p in current_puppetclasses]
+            for puppet_class_name in puppetclasses:
+                scope = {'environment_id': hostgroup['environment_id']}
+                results = module.list_resource("puppetclasses", params=scope, search='name="{0}"'.format(puppet_class_name))
+                if len(results) == 1 and len(list(results.values())[0]) == 1:
```

Comment:
This is a bit hazy. What do we see here on what level of the result? Maybe we should be explicit, by tearing it apart into separate apropriately named variables.